src/modular/boxed_residue/pow.rs

```diff
@@ -113,6 +113,7 @@ fn pow_montgomery_form(
     // Ensure output is fully reduced (AMM only reduces to the bit length of the modulus)
     // See RustCrypto/crypto-bigint#441
     z.conditional_sbb_assign(modulus, !z.ct_lt(modulus));
+    z.conditional_sbb_assign(modulus, !z.ct_lt(modulus));
```

Comment:
Alternatively we could always compute the remainder and then select between the original `z` and the remainder based on `z.ct_lt(modulus)`, skipping the conditional subtractions, but that would add allocations and as things currently stand a decent performance hit if we were to use constant-time remainder.